htop.c

```diff
@@ -137,9 +139,33 @@ static CommandLineSettings parseArguments(int argc, char** argv) {
             }
 
             while(pid) {
-                unsigned int num_pid = atoi(pid);
-                Hashtable_put(flags.pidWhiteList, num_pid, (void *) 1);
-                pid = strtok_r(NULL, ",", &saveptr);
+               char *endptr;
+               unsigned long num_pid;
+
+               if (pid[0] == '-') {
+                  fprintf(stderr, "Error: PID is negative \"%s\".\n", pid);
```

Comment:
These fprintf's are in spirit of the other option handling when in the case of malformed input a message is printed to stderr.